src/Uno.UI/UI/Xaml/Automation/AutomationProperties.cs

```diff
@@ -148,6 +149,13 @@ private static void OnAutomationIdChanged(DependencyObject dependencyObject, Dep
 			if (FrameworkElementHelper.IsUiAutomationMappingEnabled && dependencyObject is UIElement uiElement)
 			{
 				uiElement.SetAttribute("xamlautomationid", (string)args.NewValue);
+				uiElement.SetAttribute("aria-label", (string)args.NewValue);
+
+				var role = FindHtmlRole(uiElement);
+				if (role != null)
+				{
+					uiElement.SetAttribute("role", role);
```

Comment:
Is it necessary to set `aria-label` if role is null ?

Reply:
In those cases, we might not have the full requirement (role + aria-label) but it would make it easy to find for errors in the future. Meaning: `arial-label` available but no `role` it's a matter of the casing.

Reply:
@jeromelaban  updated the PR to set the `arial-label` only when a `role` is found. This way we can use the method overload suggested by @carldebilly. Two birds one shot. :)